src/test/java/io/cdap/delta/bigquery/BigQueryEventConsumerTest.java

```diff
@@ -482,6 +482,46 @@ public void testSoftDeletes() throws Exception {
     }
   }
 
+  @Test
+  public void testBigQuerySchemaNormalization() throws Exception {
+    String bucketName = "bqtest-" + UUID.randomUUID().toString();
+    Bucket bucket = storage.create(BucketInfo.of(bucketName));
+
+    BigQueryEventConsumer eventConsumer = new BigQueryEventConsumer(MockContext.INSTANCE, storage, bigQuery, bucket,
+        project, 0, STAGING_TABLE_PREFIX, false, null, null,
+        null, false);
+
+    String dataset = "testSchemaNormalization";
+    String tableName = "test-table";
+    String normalizedTableName = "test_table";
+    String fieldName = "field-with-dash";
+    String normalizedFieldName = "field_with_dash";
+
+    try {
+      Schema schema = Schema.recordOf(tableName,
+          Schema.Field.of(fieldName, Schema.of(Schema.Type.INT)));
+      bigQuery.create(DatasetInfo.newBuilder(dataset).build());
+      DDLEvent createTable = DDLEvent.builder()
+          .setOperation(DDLOperation.Type.CREATE_TABLE)
```

Comment:
Can you also check if other DDL operation passes? You can modify the existing tests, i.e, add a bad field name, to achieve that.